gcp_io/gcp_interface.py

```diff
@@ -151,6 +153,27 @@ def get_md5sum(self, file: str) -> str:
             md5_hash = md5sum(file)
         return md5_hash
 
+    def check_sum(self, gcs_file: str, local_file: str) -> bool:
+        
+        """! Check if there are differences between a local file and one in GCP
+        @param gcs_file (str) Full path to the file in cloud storage.
+        @param local_file (str) Full path to the local file.
+        """
+        # If file exists, check if it has different hash
+        if os.path.exists(local_file):
+            file_content = self.get_blob(gcs_file).download_as_string()
+            remote_filehash = hashlib.sha256(file_content).hexdigest()
```

Comment:
This does not make much sense because you are downloading the flle anyway. See how we are doing it in Kronos https://github.com/kiwicampus/Kronos-Project/blob/develop/rover/ros2/src/python_utils/python_utils/io.py#L326

Reply:
```python
def get_md5sum(file: str) -> str:
    """! Gets md5sum of file
    @param file (str) Full path to file
    @returns (str) md5sum
    """
    if "gs://" in file:
        md5_hash = binascii.hexlify(
            base64.urlsafe_b64decode(get_gcp_blob(file).md5_hash)
        ).decode("utf-8")
    else:
        md5_hash = md5sum(file)
    return md5_hash
  ````